numpy/_core/tests/test_arrayobject.py

```diff
@@ -31,3 +31,39 @@ def test_matrix_transpose_equals_swapaxes(shape):
     tgt = np.swapaxes(arr, num_of_axes - 2, num_of_axes - 1)
     mT = arr.mT
     assert_array_equal(tgt, mT)
+
+
+@pytest.mark.parametrize("subclass", [False, True])
+def test_array_wrap(subclass):
+    # NumPy should allow `__array_wrap__` to be called on arrays, it's logic
+    # is designed in a way that:
+    #
+    # * Subclasses never return scalars by default (to preserve their
+    #   information).  They can choose to if they wish.
+    # * NumPy returns scalars, if `return_scalar` is passed as True to allow
+    #   manual calls to `arr.__array_wrap__` to do the right thing.
+
+    class MyArr(np.ndarray):
+        def __array_wrap__(self, arr, context=None, return_scalar=None):
+            return super().__array_wrap__(arr, context, return_scalar)
+
+    arr = np.arange(3)
+    if subclass:
+        arr = arr.view(MyArr)
+
+    arr0d = np.array(3, dtype=np.int8)
+    # Third argument not passed, None, or True "decays" to scalar.
```

Comment:
Copy & paste error from the scalar test? For an array, only `True` causes decay to scalar. Maybe just
```
# With third argument True, ndarray allows "decay" to scalar.
```